modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFClaimForm/components/MDFClaimPage/schema/yup.ts

```diff
@@ -169,11 +141,138 @@ const claimSchema = object({
 								}
 							),
 				}),
-				metrics: string().max(
-					350,
-					'You have exceeded the character limit'
+				listOfQualifiedLeads: mixed()
+					.when('selected', {
+						is: (selected: boolean) => selected,
+						then: (schema) =>
+							schema
+								.test(
+									'fileSize',
+									validateDocument.fileSize.message,
+									(listOfQualifiedLeads) => {
+										if (
+											listOfQualifiedLeads &&
+											!listOfQualifiedLeads.id
+										) {
+											return listOfQualifiedLeads
+												? Math.ceil(
+														listOfQualifiedLeads.size /
+															1000
+												  ) <=
+														validateDocument
+															.fileSize.maxSize
+												: false;
+										}
+
+										return true;
+									}
+								)
+								.test(
+									'fileType',
+									validateDocument.listOfLeadsDocuments
+										.message,
+									(listOfQualifiedLeads) => {
+										if (
+											listOfQualifiedLeads &&
+											!listOfQualifiedLeads.id
+										) {
+											return listOfQualifiedLeads
+												? validateDocument.listOfLeadsDocuments.types.includes(
+														listOfQualifiedLeads.type
+												  )
+												: false;
+										}
+
+										return true;
+									}
+								),
+					})
+					.when(['selected', 'typeActivity'], {
+						is: (
+							selected: boolean,
+							typeActivity: LiferayPicklist
+						) =>
+							(selected &&
+								typeActivity.key === TypeActivityKey.EVENT) ||
+							(selected &&
+								typeActivity.key ===
+									TypeActivityKey.MISCELLANEOUS_MARKETING),
+						then: (schema) => schema.required('Required'),
+					}),
+				metrics: string().when(['selected', 'typeActivity'], {
+					is: (selected: boolean, typeActivity: LiferayPicklist) =>
+						selected &&
+						typeActivity.key === TypeActivityKey.DIGITAL_MARKETING,
+					then: (schema) =>
+						schema
+							.required('Required')
+							.max(350, 'You have exceeded the character limit'),
+				}),
+				proofOfPerformance: object().when(
+					['selected', 'typeActivity'],
+					(selected: boolean, typeActivity) => {
+						let targetFields = {};
+
+						if (selected) {
+							switch (typeActivity.key) {
+								case TypeActivityKey.EVENT:
+									targetFields = {
+										...getEventInvitationsValidation,
+										...getEventPhotosValidation,
+										...getEventCollateralsValidation,
+									};
+									break;
+								case TypeActivityKey.DIGITAL_MARKETING:
+									targetFields = {
+										...getAllContentsFieldsValidation,
+									};
+									break;
+								case TypeActivityKey.CONTENT_MARKETING:
+									targetFields = {
+										...getAllContentsFieldsValidation,
+									};
+									break;
+								default:
+									targetFields = {
+										...getAllContentsFieldsValidation,
+										...getImagesValidation,
+									};
+									break;
+							}
+						}
```

Comment:
```suggestion
									targetFields = {
										...eventInvitationsValidation,
										...eventPhotosValidation,
										...eventCollateralsValidation,
									};

									break;
								case TypeActivityKey.DIGITAL_MARKETING:
									targetFields = allContentsFieldsValidation;

									break;
								case TypeActivityKey.CONTENT_MARKETING:
									targetFields = allContentsFieldsValidation;

									break;
								default:
									targetFields = {
										...allContentsFieldsValidation,
										...imagesValidation,
									};

									break;
							}
						}
```

Just use the spread operation just when necessary